http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/Boot.scala

```diff
@@ -751,6 +751,9 @@ object Boot extends IOApp {
         // (for example HttpJupyterDAO). Otherwise the proxyResolver falls back to default
         // hostname resolution, so it's okay to use for all clients.
         .withCustomDnsResolver(dnsResolver)
+        .withConnectTimeout(30 seconds)
```

Comment:
Defaults (and previous settings) were:
- connect timeout: 10 seconds
- request timeout: 45 seconds
- max total connections: 10

Reply:
In particular I wonder if we were bumping against the connect timeout/max connections

Reply:
Just an thought, but I wonder if we can track these metrics somehow? It would be interesting to see if the max connections is truly the issue, and why we've been seeing this error more frequently recently

Reply:
Great idea, we should have metrics for total/failed calls. At least we do for the Sam client -- not sure about WSM. Will look.

Reply:
Looks like we're not counting individual WSM APIs in metrics, but we have metrics around total requests, abnormal responses, and response time through the http4s client. 

I just added WSM charts to the `Internal Performance` section of the Leo dashboard: https://grafana.dsp-devops.broadinstitute.org/d/470LS8A7k/leonardo?orgId=1&from=now-24h&to=now

Reply:
Seems like traffic to WSM is pretty low, but there are some blips of abnormal responses